### PR TITLE
EKF: Convert sideslip fusion to use SymPy generated auto-code

### PR DESCRIPTION
**Pull request by priseborough**
Results from before and after replay of a SITL gazebo_plane log show no visual differences:

Before:
old

After:
new

Test program comparing matlab and SymPy observation Jacobians and Kalman gains passes with differences within range expected for single precision calculations:

Pass: Sideslip Hfusion max diff fraction = 2.223303e-07
Pass: Sideslip Kfusion max diff fraction = 3.751102e-07

### CODE REVIEW

File: EKF/sideslip_fusion.cpp [hunk 1]
```diff
@@ -70,49 +69,71 @@ void Ekf::fuseSideslip()
...
-		H_BETA(0) = SH_BETA[5]*SH_BETA[8] - SH_BETA[1]*SH_BETA[4]*SH_BETA[9];
-		H_BETA(1) = SH_BETA[5]*SH_BETA[10] - SH_BETA[1]*SH_BETA[4]*SH_BETA[11];
-		H_BETA(2) = SH_BETA[5]*SH_BETA[11] + SH_BETA[1]*SH_BETA[4]*SH_BETA[10];
-		H_BETA(3) = - SH_BETA[5]*SH_BETA[9] - SH_BETA[1]*SH_BETA[4]*SH_BETA[8];
-		H_BETA(4) = - SH_BETA[5]*(SH_BETA[12] - 2.0f*q1*q2) - SH_BETA[1]*SH_BETA[4]*SH_BETA[7];
-		H_BETA(5) = SH_BETA[6] - SH_BETA[1]*SH_BETA[4]*(SH_BETA[12] + 2.0f*q1*q2);
-		H_BETA(6) = SH_BETA[5]*(2.0f*q0*q1 + 2.0f*q2*q3) + SH_BETA[1]*SH_BETA[4]*(2.0f*q0*q2 - 2.0f*q1*q3);
-		H_BETA(22) = SH_BETA[5]*(SH_BETA[12] - 2.0f*q1*q2) + SH_BETA[1]*SH_BETA[4]*SH_BETA[7];
-		H_BETA(23) = SH_BETA[1]*SH_BETA[4]*(SH_BETA[12] + 2.0f*q1*q2) - SH_BETA[6];
+		float R_BETA = sq(_params.beta_noise); // observation noise variance
```
Comment (kamilritz) [inline EKF/sideslip_fusion.cpp:72]:
```suggestion
		const float R_BETA = sq(_params.beta_noise); // observation noise variance
```

File: EKF/sideslip_fusion.cpp [hunk 2]
```diff
@@ -193,36 +170,46 @@ void Ekf::fuseSideslip()
...
+				Kfusion(row) = HK52*(HK28*P(row,22) - HK28*P(4,row) + HK32*P(row,23) - HK32*P(5,row) + HK33*P(6,row) + HK34*P(2,row) + HK35*P(1,row) - HK36*P(3,row) + HK37*P(0,row));
+			}
+
+		}
+
+		Kfusion(22) = HK45*HK52;
+		Kfusion(23) = HK42*HK52;
+
+		// synthetic sideslip measurement sample has passed check so record it
+		_time_last_beta_fuse = _time_last_imu;
```
Comment (kamilritz) [inline EKF/sideslip_fusion.cpp:207]: `_time_last_beta_fuse` is going to be updated once the numerical health checks are passing.

File: EKF/sideslip_fusion.cpp (hunk 1 above)
Comment (priseborough) [reply EKF/sideslip_fusion.cpp]: done

File: EKF/sideslip_fusion.cpp (hunk 2 above)
Comment (priseborough) [reply EKF/sideslip_fusion.cpp]: Fixed

**Pull request merged**